app/renderer/components/menubar.js

```diff
@@ -0,0 +1,216 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const React = require('react')
+const Immutable = require('immutable')
+const ImmutableComponent = require('../../../js/components/immutableComponent')
+const windowActions = require('../../../js/actions/windowActions')
+const separatorMenuItem = require('../../common/commonMenu').separatorMenuItem
+const keyCodes = require('../../../js/constants/keyCodes')
+const { wrappingClamp } = require('../../common/lib/formatUtil')
+
+const showContextMenu = (rect, submenu, activeFrame) => {
+  windowActions.setContextMenuDetail(Immutable.fromJS({
+    left: rect.left,
+    top: rect.bottom,
+    template: submenu.map((submenuItem) => {
+      if (submenuItem.type === separatorMenuItem.type) {
+        return submenuItem
+      }
+      submenuItem.click = function (e) {
+        e.preventDefault()
+        if (activeFrame && activeFrame.has('key')) {
+          // Send focus back to the active web frame
+          const results = document.querySelectorAll('webview[data-frame-key="' + activeFrame.get('key') + '"]')
+          if (results.length === 1) {
+            results[0].focus()
+          }
+        }
+        windowActions.clickMenubarSubmenu(submenuItem.label)
+      }
+      return submenuItem
+    })
+  }))
+}
+
+class MenubarItem extends ImmutableComponent {
+  constructor () {
+    super()
+    this.onClick = this.onClick.bind(this)
+    this.onMouseOver = this.onMouseOver.bind(this)
+  }
+  onClick (e) {
+    if (e && e.stopPropagation) {
+      e.stopPropagation()
+    }
+    // If clicking on an already selected item, deselect it
+    const selected = this.props.menubar.props.selectedLabel
+    if (selected && selected === this.props.label) {
+      windowActions.setContextMenuDetail()
+      windowActions.setMenubarSelectedLabel()
+      return
+    }
+    // Otherwise, mark item as selected and show its context menu
+    windowActions.setMenubarSelectedLabel(this.props.label)
+    const rect = e.target.getBoundingClientRect()
+    showContextMenu(rect, this.props.submenu, this.props.activeFrame)
+  }
+  onMouseOver (e) {
+    const selected = this.props.menubar.props.selectedLabel
+    if (selected && selected !== this.props.label) {
+      this.onClick(e)
+    }
+  }
+  render () {
```

Comment:
Currently you aren't respecting the `visible: false` for menu templates which makes find next / find prev show up int he Edit menu as items but not work. These should be hidden but the shortcuts should still work.